tests/components/samsungtv/test_media_player.py

```diff
@@ -170,61 +118,33 @@ def mock_now():
 
 async def setup_samsungtv(hass, config):
     """Set up mock Samsung TV."""
-    await async_setup_component(hass, "media_player", config)
+    await async_setup_component(hass, SAMSUNGTV_DOMAIN, config)
     await hass.async_block_till_done()
 
 
-async def test_setup_with_mac(hass, remote):
+async def test_setup_with_turnon(hass, remote):
     """Test setup of platform."""
     await setup_samsungtv(hass, MOCK_CONFIG)
     assert hass.states.get(ENTITY_ID)
 
 
 async def test_setup_duplicate(hass, remote, caplog):
     """Test duplicate setup of platform."""
-    DUPLICATE = {DOMAIN: [MOCK_CONFIG[DOMAIN], MOCK_CONFIG[DOMAIN]]}
+    DUPLICATE = {
+        SAMSUNGTV_DOMAIN: [
+            MOCK_CONFIG[SAMSUNGTV_DOMAIN][0],
+            MOCK_CONFIG[SAMSUNGTV_DOMAIN][0],
+        ]
+    }
     await setup_samsungtv(hass, DUPLICATE)
-    assert "Ignoring duplicate Samsung TV fake" in caplog.text
+    assert hass.states.get(ENTITY_ID)
+    assert len(hass.states.async_all()) == 1
```

Comment:
@MartinHjelmare I think this failing test is a race condition. When passing both configs at once it fails, when using two `setup_samsungtv` calls it's working. What can I do to fix this?

Reply:
Maybe we need to block till done twice?

Reply:
I think we need to block here, but how? https://github.com/home-assistant/home-assistant/blob/d935ef1b864d2d0b8070e1546f314fdc7a577fcd/homeassistant/components/samsungtv/__init__.py#L29-L37

Reply:
We block in the test until all tasks are done.

Reply:
Sorry, I didn't read the test failure properly. We should catch the duplicate host early already in the config schema.

Make a custom validator function that checks for duplicate hosts. See the mysensors integration for an example.